contracts/registry/ReverseRegistrar.sol

```diff
@@ -77,7 +66,32 @@ contract ReverseRegistrar {
      * @return The ENS node hash of the reverse record.
      */
     function setName(string memory name) public returns (bytes32) {
-        bytes32 node = claimWithResolver(address(this), address(defaultResolver));
+        bytes32 node = claimWithResolver(
+            address(this),
+            address(defaultResolver)
+        );
+        defaultResolver.setName(node, name);
+        return node;
+    }
+
+    /**
+     * @dev Sets the `name()` record for the reverse ENS record associated with
+     * the account provided. First updates the resolver to the default reverse
+     * resolver if necessary.
+     * Only callable by controllers
+     * @param name The name to set for this address.
+     * @return The ENS node hash of the reverse record.
+     */
+    function setNameWithController(address addr, string memory name)
```

Comment:
I think the naming of this needs some workshopping - `claimWithResolver` and `setNameWithController` use 'with' to mean entirely different things.

Reply:
Also, it'd be useful for the controller to be able to do more than just set the name. Perhaps we can just expose versions of each function with the address parameter, and make them callable only by the address or a controller?